mono/metadata/sgen-mono-ilgen.c

```diff
@@ -544,13 +544,116 @@ emit_managed_allocator_ilgen (MonoMethodBuilder *mb, gboolean slowpath, gboolean
 #endif /* MANAGED_ALLOCATION */
 }
 
+#ifndef REMOVE_ME
+int *fastbump_tlab_hit;
+int *fastbump_tlab_miss;
+#endif
+static void
+emit_managed_fastbump_allocator_ilgen (MonoMethodBuilder *mb, int instance_size)
+{
+#ifdef MANAGED_ALLOCATION
+	int p_var, tlab_next_addr_var, new_next_var, thread_var;
+	guint32 slowpath_branch;
+	MonoType *int_type = mono_get_int_type ();
+
+	const int vtable_arg = 0;
+	int size = ALIGN_TO (instance_size, SGEN_ALLOC_ALIGN);
+
+	/*
+	 * We need to modify tlab_next, but the JIT only supports reading, so we read
+	 * another tls var holding its address instead.
+	 */
+
+	EMIT_TLS_ACCESS_VAR (mb, thread_var);
+
+	/* tlab_next_addr (local) = tlab_next_addr (TLS var) */
+	tlab_next_addr_var = mono_mb_add_local (mb, int_type);
+	EMIT_TLS_ACCESS_NEXT_ADDR (mb, thread_var);
+	mono_mb_emit_stloc (mb, tlab_next_addr_var);
+
+	/* p = *(void**)tlab_next; */
+	p_var = mono_mb_add_local (mb, int_type);
+	mono_mb_emit_ldloc (mb, tlab_next_addr_var);
+	mono_mb_emit_byte (mb, CEE_LDIND_I);
+	mono_mb_emit_stloc (mb, p_var);
+
+	/* new_next = (char*)p + size; */
+	new_next_var = mono_mb_add_local (mb, int_type);
+	mono_mb_emit_ldloc (mb, p_var);
+	mono_mb_emit_icon (mb, size);
+	mono_mb_emit_byte (mb, CEE_CONV_I);
+	mono_mb_emit_byte (mb, CEE_ADD);
+
+	mono_mb_emit_stloc (mb, new_next_var);
+
+	/* if (new_next >= tlab_temp_end) */
+	mono_mb_emit_ldloc (mb, new_next_var);
+	EMIT_TLS_ACCESS_TEMP_END (mb, thread_var);
+	slowpath_branch = mono_mb_emit_short_branch (mb, MONO_CEE_BGE_UN_S);
```

Comment:
This makes sense to me, but where is MONO_CEE_BGE_UN_S defined?

Reply:
https://github.com/mono/mono/blob/b1b61d7cb7f580f5e5438a756ef3e67a5a832eeb/mono/cil/opcode.def#L54

The `MONO_` prefix is introduced by some pre-processor magic 🙄